caffe2/python/convnet_benchmarks.py

```diff
@@ -594,45 +621,60 @@ def create_resnet50_model_ops(model, loss_scale=1.0):
         brew.accuracy(model, [softmax, "label"], prefix + "_accuracy")
         return [loss]
 
-    def add_parameter_update_ops(model):
-        brew.add_weight_decay(model, weight_decay)
-        iter = brew.iter(model, "iter")
-        lr = model.net.LearningRate([iter],
-                                    "lr",
-                                    base_lr=base_learning_rate,
-                                    policy="fixed",
-                                    gamma=0.1)
-        for param in model.GetParams():
-            param_grad = model.param_to_grad[param]
-            param_momentum = model.param_init_net.ConstantFill(
-                [param], param + '_momentum', value=0.0 )
+    with core.NameScope(""):
+        with core.DeviceScope(device_opt):
+            losses = create_resnet50_model_ops(train_model)
+            if not args.forward_only:
+                blobs_to_gradients = train_model.AddGradientOperators(losses)
+                add_parameter_update_ops_resnet(train_model, base_learning_rate, weight_decay)
+        if not args.forward_only:
+            optimize_gradient_memory_resnet(train_model, [blobs_to_gradients[losses[0]]])
 
-            model.net.MomentumSGDUpdate(
-                [param_grad, param_momentum, lr, param],
-                [param_grad, param_momentum, param],
-                momentum=0.9,
-                nesterov=1)
+    return train_model, 224
 
-    def optimize_gradient_memory(model, loss):
-        model.net._net = memonger.share_grad_blobs(
-            model.net,
-            loss,
-            set(model.param_to_grad.values()),
-            namescope="",
-            share_activations=False)
+def Resnext101(args):
+    gpus = [0]
+    device_opt = core.DeviceOption(caffe2_pb2.HIP)
+    device_opt.device_id = gpus[0]
+    num_labels = 1000
+    base_learning_rate = 0.0004 * args.batch_size
+    num_layers = 101
+    num_groups = 32
+    num_width_per_group = 4
+    # Weight decay (L2 regularization)
+    weight_decay = 1e-4
 
+    ##################
+    # Define the Model
+    ##################
+    train_model = model_helper.ModelHelper(name="resnext101_train")
+
+    def create_resnext_model_ops(model, loss_scale=1.0):
+        # residual network
+        [softmax, loss] = resnet.create_resnext(model,
+                                                 "data",
+                                                 num_input_channels=3,
+                                                 num_labels=num_labels,
+                                                 label="label",
+                                                 num_layers=num_layers,
+                                                 num_groups=num_groups,
+                                                 num_width_per_group=num_width_per_group,
+                                                 no_bias=True, )
```

Comment:
Do we need no_bias true for resnext? Does the actual benchmark have it?

Reply:
Yes, the original benchmark uses that argument. https://github.com/ROCmSoftwarePlatform/pytorch/blob/676213fb2aae51b0dd95e14395c8828213722ece/caffe2/python/examples/resnet50_trainer.py#L377